API/SSW.Rewards.Application/Achievements/Command/PostAchievement/PostAchievementCommand.cs

```diff
@@ -52,11 +52,10 @@ public async Task<PostAchievementResult> Handle(PostAchievementCommand request,
 
             var userAchievements = await _context
                 .UserAchievements
-                .Where(ua => ua.UserId == user.Id)
-                .Where(ua => ua.AchievementId == achievement.Id)
```

Comment:
^^ This caused major upset with the logic below that would allow duplicate achievements to be added.